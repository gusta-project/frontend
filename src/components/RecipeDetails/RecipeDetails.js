import PropTypes from 'prop-types';
import { Table } from 'react-bootstrap';
import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export default function RecipeDetails({
  maxVg,
  percentVg,
  shakeAndVape,
  steepDays,
  flavors
}) {
  const percentPg = 100 - percentVg;
  const flavorTotal = flavors.reduce((accum, current) => {
    return accum + current.percent;
  }, 0);

  return (
    <Table striped bordered>
      <caption className="text-center">
        {maxVg ? (
          <span>Max VG |&nbsp;</span>
        ) : (
          <span>
            {percentPg}% PG / {percentVg}% VG |&nbsp;
          </span>
        )}
        {shakeAndVape ? (
          <span>Shake &amp; Vape |&nbsp;</span>
        ) : (
          <span>Steep for {steepDays} days |&nbsp;</span>
        )}
        <span>Flavor total: {flavorTotal}%</span>
      </caption>
      <thead>
        <tr>
          <th>Flavor</th>
          <th>Percent</th>
          <th>Stash</th>
        </tr>
      </thead>
      <tbody>
        {flavors.map((flavor, index) => (
          <tr key={index}>
            <td>
              <a href={'/flavor?id=' + flavor.id}>
                {flavor.abbreviation} {flavor.name}
              </a>
            </td>
            <td>{flavor.percent.toFixed(1)}%</td>
            <td>
              {flavor.inStash ? (
                <FontAwesomeIcon icon={['far', 'check-square']} />
              ) : (
                <FontAwesomeIcon icon={['far', 'square']} />
              )}
            </td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
}

RecipeDetails.defaultProps = {
  maxVg: false,
  percentVg: 50,
  shakeAndVape: false,
  steepDays: 0,
  flavors: []
};

RecipeDetails.propTypes = {
  maxVg: PropTypes.bool.isRequired,
  percentVg: PropTypes.number.isRequired,
  shakeAndVape: PropTypes.bool.isRequired,
  steepDays: PropTypes.number.isRequired,
  flavors: PropTypes.arrayOf(PropTypes.object).isRequired
};
