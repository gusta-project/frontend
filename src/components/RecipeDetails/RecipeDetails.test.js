import { render } from '@testing-library/react';
import React from 'react';

import RecipeDetails from './RecipeDetails';

describe('<RecipeDetails />', () => {
  it('renders correctly', () => {
    const flavors = [
      {
        id: 1,
        name: 'Sturgeon Ripe',
        abbreviation: 'TPA',
        percent: 1,
        inStash: true
      }
    ];
    const { asFragment } = render(
      <RecipeDetails
        maxVg={false}
        percentVg={75}
        shakeAndVape={false}
        steepDays={7}
        flavors={flavors}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
