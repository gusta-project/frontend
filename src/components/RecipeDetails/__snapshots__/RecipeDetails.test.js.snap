// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetails /> renders correctly 1`] = `
<DocumentFragment>
  <table
    class="table table-striped table-bordered"
  >
    <caption
      class="text-center"
    >
      <span>
        25% PG / 75% VG | 
      </span>
      <span>
        Steep for 7 days | 
      </span>
      <span>
        Flavor total: 1%
      </span>
    </caption>
    <thead>
      <tr>
        <th>
          Flavor
        </th>
        <th>
          Percent
        </th>
        <th>
          Stash
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            href="/flavor?id=1"
          >
            TPA Sturgeon Ripe
          </a>
        </td>
        <td>
          1.0%
        </td>
        <td>
          <i
            class="fa far-check-square"
          />
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
